### PR TITLE
SDK 3.37.1 : Adds version bump and changelog

### PR DESCRIPTION
**Pull request by NaluTripician**
# Pull Request Template

## Description

This bumps the SDK version 3.37.1. There was no changes to the GA public contract.

Preview contract changes:

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

**Review by kundadebdatta — APPROVED** [review]
LGTM!

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs [hunk 1]
```diff
@@ -345,7 +347,16 @@ private static bool IsNonRetriableException(Exception exception)
                 }
 
                 return false;
-            }
+            }
+
+            public void Dispose()
+            {
+                if (Interlocked.Increment(ref this.disposeCounter) == 1)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:354]: Is it done as safe practice? (AFAIR its not followed in other places)

**Review by kirankumarkolli — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs (hunk 1 above)
Comment (NaluTripician) [reply Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:354]: I believe so, yes. Please look at Fabian's comment on the original PR. https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4226#discussion_r1437594014

**Review by ealsur — CHANGES_REQUESTED** [review]
No code changes in version bump/release PRs

File: changelog.md [hunk 2]
```diff
@@ -13,6 +13,13 @@ Preview features are treated as a separate branch and will not be included in th
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### <a name="3.37.1-preview"/> [3.37.1-preview](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.37.1-preview) - 2024-1-2
+### <a name="3.37.1"/> [3.37.1](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.37.1) - 2024-1-2
+
```
Comment (ealsur) [inline changelog.md:18]: nit: remove empty line

**Review by ealsur — APPROVED** [review]

**Review by ealsur — APPROVED** [review]

**Review by kundadebdatta — APPROVED** [review]

**Pull request merged**